### PR TITLE
Feature/ssl support

### PR DESCRIPTION
**Pull request by mindriot88**
## Pull Request Checklist

* Add the ability to set the `scheme` redis client option which opens up support for redis server configurations which use SSL.

This can be set on any of the plugin checks using `-T rediss` or `--transport rediss`

#### General

- [ ] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

**Comment by majormoses** [comment]
Hmm looking at the code I would not have expected that to break the integration tests. I will try to take a closer look when I have more time.

**Comment by majormoses** [comment]
Im thinking `scheme` is not the right config option name. It is ambiguous and often conflicts with metric gathering scripts that offer a `--scheme my.hostname.my_custom_service` to control how the metrics are formatted before sending to your TSDB. I would opt for `transport` or `protocol`. Also is there a list of protocols that we can restrict it to? if so we can do something similar to this: https://github.com/sensu-plugins/sensu-plugins-redis/blob/3.0.1/lib/redis_client_options.rb#L75 so it helps users know what the options are.

**Comment by mindriot88** [comment]
@majormoses I will take a look at revising the implementation based on your feedback once I get a free moment, thanks

**Comment by mindriot88** [comment]
The options that it should be restricted to are `redis` and `rediss` as those are the only supported schemes in the redis client, I will make those changes.

**Comment by mindriot88** [comment]
@majormoses your comments should be addressed, ive added a test as well because wasn't able to run your kitchen tests locally, I see they are failing in travis as well at the moment.

Let me know if you need anything else changing once those are fixed.

**Comment by majormoses** [comment]
Overall looks good but I need to look into why the tests are failing.

Assuming you have a working ruby and docker setup you should be able to run the tests via:
```
$ bundle install
...
$ bundle exec kitchen converge ruby-241-debian-8
```

File: lib/redis_client_options.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def default_redis_options
     opts[:password] = config[:password] if config[:password]
     opts[:timeout]  = config[:timeout]  if config[:timeout]
     opts[:db]       = config[:database] if config[:database]
+    opts[:scheme]   = config[:transport] if config[:transport]
```
Comment (majormoses) [inline lib/redis_client_options.rb]: looks like this needs to be changed to `opts[:transport]`

File: test/lib/redis_client_options_spec.rb [hunk 2]
```diff
@@ -42,4 +42,11 @@ def run
     check = DummyRedisCheck.new(args)
     expect(check.redis_endpoint).to eq 'unix:///some/path/redis.sock'
   end
+
+  it 'sets transport protocol' do
+    args = %w[--transport redis]
+    check = DummyRedisCheck.new(args)
+    expect(check.config[:transport]).to eq 'redis'
+    expect(check.default_redis_options[:scheme]).to eq 'redis'
```
Comment (majormoses) [inline test/lib/redis_client_options_spec.rb]: this should be `:transport`

File: lib/redis_client_options.rb [hunk 3]
```diff
@@ -68,6 +69,14 @@ def _configure_options
              long: '--password PASSWORD',
              description: 'Redis Password to connect with'
 
+      option :transport,
+             short: '-T TRANSPORT',
+             long: '--transport TRANSPORT',
+             description: 'Redis transport protocol to use',
+             required: false,
+             default: Redis::Client::DEFAULTS[:scheme],
```
Comment (majormoses) [inline lib/redis_client_options.rb]: this should be transport

**Comment by majormoses** [comment]
I am able to replicate the tests failing locally, not sure if they are bad tests or if there was some other change.

**Comment by majormoses** [comment]
This one looks to me like maybe the output changed in redis versions? Need to validate and if so we can make it match either one.
```
1) Command "/usr/local/bin/check-redis-slave-status.rb -p 6380" stdout should match /The\ redis\ master\ links\ status\ is\ up!/
            Failure/Error: its(:stdout) { should match(Regexp.new(Regexp.escape('The redis master links status is up!'))) }
              expected "RedisSlaveCheck OK: This redis server is master\n" to match /The\ redis\ master\ links\ status\ is\ up!/
              Diff:
              @@ -1,2 +1,2 @@
              -/The\ redis\ master\ links\ status\ is\ up!/
              +RedisSlaveCheck OK: This redis server is master
```

**Comment by majormoses** [comment]
This looks like maybe its a legitimate problem or something with how we are setting up the tests as there should be nothing listening on `6381`:
```
2) Command "/usr/local/bin/check-redis-slave-status.rb -p 6381" exit_status should eq 2
            Failure/Error: its(:exit_status) { should eq 2 }
              
              expected: 2
                   got: 0
              
              (compared using ==)
              /bin/sh -c /usr/local/bin/check-redis-slave-status.rb\ -p\ 6381
              RedisSlaveCheck OK: This redis server is master
```

**Comment by majormoses** [comment]
I was able to validate those failures on master.

**Comment by majormoses** [comment]
Looks like the last passing build (and commit) on master was 8 months ago: https://travis-ci.org/sensu-plugins/sensu-plugins-redis/builds/359647878 so its probably something that changed in redis and those tests need to be updated. I am gonna be traveling this week for the holidays so I will probably take care of this when I get back.

File: lib/redis_client_options.rb (hunk 1 above)
Comment (mindriot88) [reply lib/redis_client_options.rb]: correct me if I am wrong but are these not the options passed to the redis client, in which `scheme` is the correct option we are trying to set?

Comment (majormoses) [reply lib/redis_client_options.rb]: ah I had not looked at this code for a while and misunderstood based on the diff github showed me.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-redis/versions/3.1.0

**Pull request merged**